### PR TITLE
refactor(react-dom): wrap all Kinase functionality in react

### PR DESCRIPTION
**Pull request by jrbotros**
Also adds open-close functionality to Kinase sidebar in preparation for rendering the extension on every page.

screen shot 2017-05-16 at 6 54 23 pm
screen shot 2017-05-16 at 6 53 58 pm

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -14,5 +14,8 @@ module.exports = {
 
     // TODO(jrbotros): assign collection items a unique identifier
```
Comment (marcua) [inline .eslintrc.js]: how does someone turn off rendering/running this code?

File: src/components/Highlighter/index.jsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  clear() {
+    if (!isUndefined(this.tether)) {
+      this.tether.target.classList.remove(styles.kinaseHighlightTarget);
+      this.tether.destroy();
+    }
+  }
+
+  highlight(target) {
+    this.element.style.width = `${target.offsetWidth + this.padding}px`;
```
Comment (marcua) [inline src/components/Highlighter/index.jsx]: can this and `this.padding` move into scss?

File: .eslintrc.js [hunk 3]
```diff
@@ -14,5 +14,8 @@ module.exports = {
```
Comment (marcua) [inline .eslintrc.js]: err bottom left since it swings out from the left

Comment (meredithblumenstock) [inline .eslintrc.js]: @marcua @jrbotros Bottom left is a good idea.

Comment (jrbotros) [inline .eslintrc.js]: Going to leave as-is for now unless you feel strongly. My next PR will allow the user to toggle-off the extension entirely (so the tab won't even show up).

File: .eslintrc.js (hunk 1 above)
Comment (jrbotros) [inline .eslintrc.js]: Refreshing the page currently—this will be addressed in the next PR.

**Pull request merged**